src/controllers/providers/providers.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable no-underscore-dangle */
+import { Network, NetworkId } from '../../interfaces/network'
+import { RPCProviders } from '../../interfaces/provider'
+import { getRpcProvider } from '../../services/provider'
+import EventEmitter from '../eventEmitter/eventEmitter'
+import { NetworksController } from '../networks/networks'
+
+export class ProvidersController extends EventEmitter {
```

Comment:
Please when you have some time, add a few notes for this controller (that sums up its purpose).